### DISCUSSION TITLE
is WinUI3 ready for something like Office?

### DISCUSSION THREAD

**Discussion by meirumeiru** [microsoft/WindowsAppSDK#2283] [General]
Do you think, WinUI3 is ready to support a product like Office? ... or do you think it will take some time (2023? 2024?) until we should think about using it in such a project?

**Comment by ghost** [comment]
> Do you think, WinUI3 is ready to support a product like Office? 

Absolutely Not. 

> .. or do you think it will take some time (2023? 2024?)

2024-2025 will be a good guess.

Why ? See https://github.com/microsoft/microsoft-ui-xaml/issues/1500#issuecomment-1033409077 and then https://github.com/microsoft/microsoft-ui-xaml/issues/6791#issuecomment-1065328532

**Comment by kmgallahan** [comment]
WinUI is capable of drawing anything on the screen, so essentially any program can be created - except for those requiring some of the Windows API surface that isn't available yet. [Productboard](https://portal.productboard.com/winappsdk/1-windows-app-sdk/tabs/2-planned)

The question is how much time and willpower you want to exert making something. Case in point, anyone *could* create a datagrid (in C++, there are several C# versions available [1](https://www.syncfusion.com/winui-controls/datagrid), [2](https://www.telerik.com/winui/datagrid), [3](https://docs.microsoft.com/en-us/windows/communitytoolkit/controls/datagrid)). Don't forget that Office was already recreated once using UWP (including Excel) - it didn't do everything the original could, but that isn't something that more time and money wouldn't solve...
